src/db/sysdb_sudo.c

```diff
@@ -76,9 +76,18 @@ static errno_t sysdb_sudo_convert_time(const char *str, time_t *unix_time)
     for (format = formats; *format != NULL; format++) {
         /* strptime() may leave some fields uninitialized */
         memset(&tm, 0, sizeof(struct tm));
+        /* Let underlying implementation figure out DST */
+        tm.tm_isdst = -1;
```

Comment:
Is there any reason why you are setting it to `-1`?

Reply:
bc apparently this is the correct way to init the struct tm. if you use any API like mktime(), which imply broken down time expressed as local time, when you set tm_isdst to zero you are explicitly telling it DST is not in effect even before the structure is set with any actual data. on API like mktime() this could lead to unexpected results eg time jumps of +-1h vs what is expected.

note however that in the case of this patch it does not seem to matter for timegm() probably bc DST is not applicable to UTC/GMT time and it just ignores it so tm_isdst can be set to anything but i chose to set it to "undefined" anyway just to keep things clean since its literally just one instruction and doesnt really worth skipping. so while technically its not required in this case i reckon it is better to init tm_isdst as "undefined".

https://www.gnu.org/software/libc/manual/html_node/Broken_002ddown-Time.html
https://pubs.opengroup.org/onlinepubs/9699919799/functions/mktime.html

Reply:
Thanks for the clarification and the links